org.eclipse.draw2d/src/org/eclipse/draw2d/EventListenerList.java

```diff
@@ -104,8 +59,9 @@ public void remove() {
 	 * @param listenerType the type
 	 * @return an Iterator of all the listeners of type <i>c</i>
 	 */
+	@SuppressWarnings("unchecked")
 	public synchronized <T> Iterator<T> getListeners(final Class<T> listenerType) {
-		return new TypeIterator<>(array, listenerType);
+		return (Iterator<T>) listeners.getOrDefault(listenerType, Collections.emptyList()).iterator();
```

Comment:
Unlike before, this iterator now supports the "remove" operation. Is this desirable or something we would have to actively suppress?

Reply:
With the new design it wouldn't matter to much. I think it was just not allowed as in the old implementation it would have been extremly hard to implement.